src/coord/src/catalog.rs

```diff
@@ -917,6 +1012,22 @@ impl Catalog {
                     }
                     OpStatus::DroppedItem(metadata)
                 }
+
+                Action::UpdateItem { id, name, item } => {
+                    let mut entry = self.by_id.remove(&id).unwrap();
```

Comment:
Since we're not updating the uses/used_by metadata here, it'd be swell if you could add an assertion that asserts the `UpdateItem` isn't changing the item's uses:

```
assert_eq!(entry.uses(), item.uses());
```